src/databricks/labs/remorph/reconcile/connectors/data_source.py

```diff
@@ -1,34 +1,48 @@
+import logging
 import re
 from abc import ABC, abstractmethod
 
 from pyspark.sql import DataFrame, SparkSession
 
+from databricks.labs.remorph.config import TableRecon
 from databricks.labs.remorph.reconcile.recon_config import JdbcReaderOptions, Schema
 from databricks.sdk import WorkspaceClient
 
+logger = logging.getLogger(__name__)
+
 
 class DataSource(ABC):
     # TODO need to remove connection_params
     def __init__(
         self,
-        engine: str,
         spark: SparkSession,
         ws: WorkspaceClient,
         scope: str,
+        engine: str = None,
```

Comment:
```suggestion
        engine: Dialect,
```

not blocking this PR, but it's a code smell when you have to be passing down the instance of `sqlglot.dialects.dialect.Dialect` and instead pass a string to reference it. this will blow up very soon.

also, you have `SparkSession` and `WorkspaceClient` as "global object instances" passed via their instances, so be consistent with that.